cairo/ethereum/cancun/vm/instructions/environment.cairo

```diff
@@ -0,0 +1,59 @@
+from starkware.cairo.common.bitwise import BitwiseBuiltin
+from starkware.cairo.common.math import split_felt
+
+from ethereum_types.bytes import Bytes32, Bytes32Struct
+from ethereum_types.numeric import U256, Uint, U256Struct
+
+from ethereum.cancun.vm import Evm, EvmImpl
+from ethereum.cancun.vm.exceptions import ExceptionalHalt
+from ethereum.cancun.vm.gas import charge_gas, GasConstants
+from ethereum.cancun.vm.stack import pop, push
+
+from ethereum.utils.numeric import U256_lt, U256_from_be_bytes
+
+// @notice Gets the versioned hash at a particular index
+func blob_hash{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, evm: Evm}() -> ExceptionalHalt* {
+    alloc_locals;
+
+    // STACK
+    let stack = evm.value.stack;
+    with stack {
+        let (index, err) = pop();
+        if (cast(err, felt) != 0) {
+            return err;
+        }
+    }
+
+    // GAS
+    let err = charge_gas(Uint(GasConstants.GAS_BLOBHASH_OPCODE));
+    if (cast(err, felt) != 0) {
+        return err;
+    }
+
+    // OPERATION
+    let blob_hashes = evm.value.env.value.blob_versioned_hashes;
+
+    // If index is within bounds, get the hash at that index
+    // Otherwise return zero bytes
+    let (high, low) = split_felt(blob_hashes.value.len);
```

Comment:
i guess it's fine to do this because we don't want to over analyze the evm

but blob_hashes.value.len right now is between 2 and 5, in the future it'll be at most 200 or maybe 1000